dateparser/__init__.py

```diff
@@ -7,7 +7,8 @@
 
 
 @apply_settings
-def parse(date_string, date_formats=None, languages=None, locales=None, region=None, settings=None):
+def parse(date_string, date_formats=None, languages=None, locales=None,
+          region=None, settings=None, detect_languages_func=None):
```

Comment:
sorry if this was already discussed and I missed it - do we need a new argument ``detect_languages_func``, can this go into settings instead?

Reply:
It was in setting but we decided to make it independent of settings.

Reply:
I don't think a good idea to add a function definition in the settings. For me it's totally fine to have this argument. Maybe we can impove the naming, but the parameter it's OK.

Reply:
Aha I see - sounds good to me. On the meeting we also discussed whether we could still put that into setting, maybe via two arguments (a string one for built-in detectors and a function one for custom), but I'm not sure which is better.

Reply:
This idea is great. The only issue I feel is that it will make usage of the library complex.

Reply:
Hi @noviluni,
Can you provide your suggestions?